test/controllers/power_controller_test.rb

```diff
@@ -0,0 +1,210 @@
+      post "#{samples_url}.json", params: { sample: { name: 'joe new sample', project_name: @joe_project.name, input_files_attributes: input_files } }
+    end
+    assert_response :success
+  end
+
+  test 'joe can update sample to joe_project' do
+    @joe_sample = samples(:joe_sample)
+    post "#{save_metadata_sample_url(@joe_sample)}.json", params: { field: 'sample_tissue', value: 'bone' }
+    assert_response :success
+    @joe_sample.reload
+    assert @joe_sample.sample_tissue == 'bone'
+  end
+
+  test 'joe can see samples  in joe_project' do
+    @joe_project = projects(:joe_project)
+    get "/samples.json?project_id=#{@joe_project.id}"
+    assert_response :success
+    assert JSON.parse(@response.body)["total_count"] == 1
+  end
+
+  test 'joe can see joe_sample' do
+    @joe_sample = samples(:joe_sample)
+    get sample_url(@joe_sample)
+    assert_response :success
+  end
+
+  # public projects
+
+  test 'joe cannot add users to public_project ' do
+    @public_project = projects(:public_project)
+    assert_raises(ActiveRecord::RecordNotFound) do
+      put add_user_project_url(@public_project), params: { user_email_to_add: "abc@xyz.com"  }
+    end
+  end
+
+  test 'joe cannot change project visibility to public_project ' do
+    @public_project = projects(:public_project)
+    assert_raises(ActiveRecord::RecordNotFound) do
+      put update_project_visibility_project_url(@public_project), params: { public_access: 0 }
+    end
+  end
+
+  test 'joe cannot create sample to public_project' do
+    @public_project = projects(:public_project)
+    input_files = [{ source: "RR004_water_2_S23_R1_001.fastq.gz",
+                     name: "RR004_water_2_S23_R1_001.fastq.gz",
+                     source_type: "local" },
+                   { source: "RR004_water_2_S23_R2_001.fastq.gz",
+                     name: "RR004_water_2_S23_R2_001.fastq.gz",
+                     source_type: "local" }]
+    assert_no_difference('Sample.count') do
+      post "#{samples_url}.json", params: { sample: { name: 'joe new sample', project_name: @public_project.name, input_files_attributes: input_files } }
+    end
+    assert_response 422
+  end
+
+  test 'joe cannot update sample to public_project' do
+    @public_sample = samples(:public_sample)
+    assert_raises(ActiveRecord::RecordNotFound) do
+      post "#{save_metadata_sample_url(@public_sample)}.json", params: { field: 'sample_tissue', value: 'bone' }
+    end
+    @public_sample.reload
+    assert @public_sample.sample_tissue != 'bone'
+  end
+
+  test 'joe can see samples in public_project' do
+    @public_project = projects(:public_project)
+    get "/samples.json?project_id=#{@public_project.id}"
+    assert_response :success
+    assert JSON.parse(@response.body)["total_count"] == 1
+  end
+
+  test 'joe can see public_sample' do
+    @public_sample = samples(:public_sample)
+    get sample_url(@public_sample)
+    assert_response :success
+  end
+
```

Comment:
What about:

* added or 'member' user can see all samples in project

* non-member user cannot see any recent sample in private project

* non-member user can see samples in private project that are public due to privacy expiration

what metadata can be seen through privacy expiration?

Reply:
oh i see the above was for public projects sorry